source/common/http/conn_manager_utility.cc

```diff
@@ -319,6 +319,11 @@ void ConnectionManagerUtility::mutateResponseHeaders(Http::HeaderMap& response_h
     response_headers.insertRequestId().value(*request_headers.RequestId());
   }
 
+  if (protocol == Protocol::Http2) {
```

Comment:
thinking out loud: should we also remove Upgrade? it's also mentioned in the RFC

Reply:
tl;dr let's leave it as-is for now.

Per https://tools.ietf.org/html/draft-mcmanus-httpbis-h2-websockets-02 when we do upgrade over h2 connect, we won't be using the upgrade headers on the H2-H2 hop.  That said, we've had some discussion of how we want to handle normalizing websocket.   We don't want things like the CORS filter to have to look at CONNECT for H2 and GET/Upgrade:Websocket for HTTP/1.    The plan of record is to have the websocket data traverse the Envoy filter chain normalized to HTTP/1.1 so we'd turn successful H2 bootstrapped/connect/websocket into HTTP/1 Upgrade:Websocket when we read it in from the H2 codec, pass it through Envoy, and then convert it back when serializing back to H2.   So I think for this header, we'll strip it at the H2 codec.